### PR TITLE
chore(DTFS2-7532): gov design library - character count updates

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
- GOV design library has been updated.
- This PR updates the Portal's consumption of character count related inputs to adhere to the latest GOV design library.

## Resolution :heavy_check_mark:
- For text inputs that use character count (i.e not a textarea, but a text input with character count functionality):
  - Add `govuk-form-group` class to the container div.
  - Add `govuk-!-margin-bottom-1` class to the text input, to match spacing with the character count component.
  - Replace hard coded text input HTML to use `govukInput` component.
- For hard coded HTML to mimic textarea character components:
  - Replace with the `govukCharacterCount` component.

## Miscellaneous :heavy_plus_sign:
- Remove some unused nunjucks imports.
- Remove now unnecessary `govuk-js-character-count` class from `govukCharacterCount` instances.
- Simplify "bond details" E2E test with a `beforeEach`.

### CODE REVIEW

File: portal/templates/bond/bond-details.njk [hunk 1]
```diff
@@ -38,27 +39,26 @@
...
+            {{ govukCharacterCount({
+              name: "bondIssuer",
+              id: "bondIssuer",
+              value: bond.bondIssuer,
+              maxlength: 150,
+              label: {
+                text: "Bond issuer (optional)"
+              },
+              hint: {
+                text: "Only enter if Bond issuer differs from the bank"
```
Comment (Zainzzkk) [inline portal/templates/bond/bond-details.njk:53]: do we need a seperate data-cy for the hint for testing?

File: portal/templates/contract/about/components/supplier-name-and-address-lookup.njk [hunk 2]
```diff
@@ -10,25 +10,30 @@
...
+      name: "supplier-name",
       value: deal.submissionDetails["supplier-name"],
       attributes: {
         "data-cy": "supplier-name",
         "maxlength": "150"
       },
       errorMessage: validationErrors.errorList["supplier-name"]
     }) }}
+
+    <span id="supplier-name-info" class="govuk-hint govuk-character-count__message" aria-live="polite">
```
Comment (Zainzzkk) [inline portal/templates/contract/about/components/supplier-name-and-address-lookup.njk:33]: same about data-cy

File: portal/templates/contract/about/components/legally-distinct-counter-indemnifier_slash_guarantor.njk [hunk 3]
```diff
@@ -67,6 +66,10 @@
         },
         errorMessage: validationErrors.errorList["indemnifier-name"]
       }) }}
+
+      <span id="indemnifier-name-info" class="govuk-hint govuk-character-count__message" aria-live="polite">
```
[the remark this answers is not in the document]
Comment (ttbarnes) [reply portal/templates/contract/about/components/legally-distinct-counter-indemnifier_slash_guarantor.njk:70]: Out of scope - there is currently no E2E test coverage for character count.

Would like to introduce a command like `cy.assertDynamicCharacterCount` in [EXIPS](https://github.com/UK-Export-Finance/exip/blob/main/e2e-tests/commands/shared-commands/assertions/assert-dynamic-character-count.js).

I have created DTFS2-7691 to tackle this.

File: portal/templates/contract/about/components/supplier-name-and-address-lookup.njk (hunk 2 above)
Comment (ttbarnes) [reply portal/templates/contract/about/components/supplier-name-and-address-lookup.njk:33]: Same as above

File: portal/templates/bond/bond-details.njk (hunk 1 above)
Comment (ttbarnes) [reply portal/templates/bond/bond-details.njk:53]: E2E test coverage is not great for this page. Have added an E2E test for the hint.

File: portal/templates/contract/contract-clone.njk [hunk 4]
```diff
@@ -26,98 +26,80 @@
...
-              You can enter up to 30 characters
-            </span>
-          </div>
+        <div class="govuk-form-group govuk-character-count" data-module="govuk-character-count" data-maxlength="30">
+          {{ govukInput({
+            label: {
+              text: "Bank deal ID"
+            },
+            hint: {
+              text: "It's not possible to change this later."
```
Comment (Zainzzkk) [inline portal/templates/contract/contract-clone.njk:35]: needs a data-cy here too for the hint

Comment (ttbarnes) [reply portal/templates/contract/contract-clone.njk:35]: Added along with an E2E test.

**Review by Zainzzkk — DISMISSED** [review]

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**